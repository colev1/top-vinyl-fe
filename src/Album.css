ul {
  list-style-type: none;
  padding-left: 5px;
}


.vinyl-container {
  background: #B4C5E4;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.track-container {
  text-align: left;
  display: flex;
}

.track-list {
  margin-top: 40px;
}

.track {
  border-bottom: 1px solid grey;
  margin-bottom: 15px;
  position: relative;
  height: 35px;
  display: flex;
  justify-content: space-between;
}

.track-name {
  display: inline-block;
  margin: 0 20px 0 0;
  max-height: 30px;
}

.delete-track-btn {
  position: relative;
  bottom: 5px;
}

.track-duration {
  display: inline-block;
  margin: 0 0 0 12px;
  font-weight: normal;
}

.album-info {
  margin-right: 30px;
}

.album-name {
  margin: 20px 0 0 0;
  padding: 0;
  font-size: 30px;
}

.album-genre {
  margin-top: 5px;
  color: #3F4756;
}

.artist {
  margin: 10px 0 0 0 ;
  color: #3F4756;
}

.view-albums-button {
  margin-bottom: 15px;
}
