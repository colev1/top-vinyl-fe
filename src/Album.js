import React, {Component} from 'react';

class Album extends Component {
  constructor(props) {
    super(props)
    this.state = {
      album: {},
      tracks: [],
      name: '',
      duration: ''
    }
  }

  componentDidMount() {
    if(this.props.albumId) {
      this.fetchTracks()
      this.fetchAlbum()
    }
  }

  fetchTracks = () => {
    const { albumId } = this.props; 
    fetch(`http://localhost:3000/api/v1/albums/${albumId}/tracks`)
      .then(response => response.json())
      .then(tracks => this.displayTracks(tracks))
      .catch(error => console.log(error))
  }

  fetchAlbum = () => {
    const { albumId } = this.props; 
    fetch(`http://localhost:3000/api/v1/albums/${albumId}`)
      .then(response => response.json())
      .then(album => {this.setState({album})})
      .catch(error => console.log(error))
  }

  displayTracks = (tracks) => {
    this.setState({tracks})
  }

  deleteTrack = (id) => {
    fetch(`http://localhost:3000/api/v1/tracks/${id}`, {
      method: 'DELETE'
    })
    .then(response => response.json())
    .then(result => this.fetchTracks())
    .catch(error => console.log(error))
  }

  addNewTrack = (e, albumId) => {
    e.preventDefault()
    let trackRequest = {
      name: this.state.name,
      duration: this.state.duration
    }
    fetch(`http://localhost:3000/api/v1/albums/${albumId}/tracks`, {
      method: 'POST',
      body: JSON.stringify(trackRequest),
      headers: {
        "Content-Type": "application/json"
      }
    })
    .then(response => response.json())
    .then(result => this.fetchTracks())
    .catch(error => console.log(error))
  }

  handleChange = (e) => {
    const {name, value} = e.target;
    this.setState({
      [name]: value
    })
  }

  render() {
    const {album, genre, artist, id} = this.state.album;
    let tracks = this.state.tracks.map(track => {
      return (
        <li key={track.id} className='track'> 
          <h4 className='track-name'> {track.name} <span className='track-duration'> {track.duration} </span></h4>
          <button className='delete-track-btn' onClick={()=> this.deleteTrack(track.id)}> delete</button>
        </li>
      )
    })

    return (
      <div className='vinyl-container'>
        <section className='track-container'>
          <article className='album-info'>
            <h1 className='album-name'> {album} </h1>
            <p className='artist'> by: {artist} </p>
            <p className='album-genre'> genre: {genre} </p>
          </article>
          <article className='track-list'>
            <ul> {tracks} </ul>
          </article>
        </section>
        <form onSubmit={(e) => this.addNewTrack(e, id)}> add a track:
          <input placeholder='name' onChange={this.handleChange}
          name='name' 
          value={this.state.name} />
          <input placeholder='duration' 
          name='duration'
          onChange={this.handleChange} value={this.state.duration} />
          <button type='submit'> submit </button>
        </form>
        <button className='view-albums-button' onClick={() => this.props.toggleDisplay()}>View All Ablums</button>
      </div>
    )
  }
}

export default Album;