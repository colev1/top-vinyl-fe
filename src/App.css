.App {
  text-align: center;
  font-family: 'Inconsolata', monospace;
}

img {
  height: 80px;
  margin-right: 20px;
}

.title {
  /* height: 100px; */
  font-size: 44px;
  display: flex;
  justify-content: center;
  align-items: center;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
