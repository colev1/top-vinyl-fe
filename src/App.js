import React, { Component } from 'react';
import AlbumContainer from './AlbumContainer'
import './App.css';

class App extends Component {
  constructor () {
    super()
    this.state = {
      albums: [],
      error: '',
      album: '',
      genre: '',
      rating: '',
      artist: '',
      year: ''
    }
  }

  componentDidMount = () => {
    this.fetchAlbums()
  }

  fetchAlbums = () => {
    fetch('http://localhost:3000/api/v1/albums')
      .then(response => response.json())
      .then(result => this.addAlbums(result))
      .catch(error => console.log(error))
  }

  addAlbums = (albums) => {
    this.setState({
      albums
    })
  }

  deleteAlbum = (id) => {
    fetch(`http://localhost:3000/api/v1/albums/${id}`, {
      method: 'DELETE'
    })
    .then(response => response.json())
    .then(result => this.fetchAlbums())
    .catch(error => console.log(error))
  }

  handleChange = (e) => {
    const {name, value} = e.target;
    this.setState({
      [name]: value
    })
  }

  handleSubmit = (e) => {
    e.preventDefault()
    const {artist, genre, year, album} = this.state;
    const rating = parseInt(this.state.rating)
    const requestBody = {artist, genre, year, rating, album};
    fetch('http://localhost:3000/api/v1/albums', {
      method: 'POST',
      body: JSON.stringify(requestBody),
      headers: {
        "Content-Type": "application/json"
      }
    })
    .then(response => response.json())
    .then(result => console.log(result))
    .then(this.fetchAlbums())
    .catch(error => console.log(error))
  }

  render() {
    let {album, artist, year, genre, rating} = this.state;
    return (
      <div className="App">
        <h1> TOP VINYL </h1>

        <form onSubmit={(e) => this.handleSubmit(e)}> 
          add a new album:
          <input placeholder='album name' 
          name="album"
          value={album}
          onChange={this.handleChange} />
          <input placeholder='artist'
          name="artist"
          value={artist}
          onChange={this.handleChange} />
          <input placeholder='genre'
          name="genre"
          value={genre}
          onChange={this.handleChange} />
          <input placeholder='year'
          name="year"
          value={year}
          onChange={this.handleChange} />
          <input placeholder='rating'
          name="rating"
          value={rating}
          onChange={this.handleChange} />
          <button type="submit"
          > 
          add new album </button>
        </form>
        <AlbumContainer albums={this.state.albums} deleteAlbum={this.deleteAlbum}/>
      </div>
    );
  }
}

export default App;
