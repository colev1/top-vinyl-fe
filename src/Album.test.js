import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import Album from './Album.js'

describe('Album', () => {
  let mockId
  let mockToggle
  let wrapper

  beforeEach(() => {
    mockId = 1
    mockToggle = jest.fn()
    wrapper = shallow(<Album albumId={ mockId } tiggleDisplay={ mockToggle } />)
  })

  it('should match the snapshot', () => {
    
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a vinyl container', () => {
    
    expect(wrapper.find('.vinyl-container').length).toEqual(1)
  })

  it('should call fetchTrack if there is an albumid in props', () => {
    const spyFetchTracks = jest.spyOn(wrapper.instance(), 'fetchTracks')
    const spyFetchAlbum = jest.spyOn(wrapper.instance(), 'fetchAlbum')

    wrapper.instance().componentDidMount()

    expect(spyFetchTracks).toHaveBeenCalled()
    expect(spyFetchAlbum).toHaveBeenCalled()
  })
})