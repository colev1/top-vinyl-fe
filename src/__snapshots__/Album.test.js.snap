// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Album
    albumId={1}
    tiggleDisplay={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="track-container"
        >
          <article
            className="album-info"
          >
            <h1
              className="album-name"
            >
               
               
            </h1>
            <p
              className="artist"
            >
               by: 
               
            </p>
            <p
              className="album-genre"
            >
               genre: 
               
            </p>
          </article>
          <article
            className="track-list"
          >
            <ul>
               
               
            </ul>
          </article>
        </section>,
        <form
          onSubmit={[Function]}
        >
           add a track:
          <input
            name="name"
            onChange={[Function]}
            placeholder="name"
            value=""
          />
          <input
            name="duration"
            onChange={[Function]}
            placeholder="duration"
            value=""
          />
          <button
            type="submit"
          >
             submit 
          </button>
        </form>,
        <button
          className="view-albums-button"
          onClick={[Function]}
        >
          View All Ablums
        </button>,
      ],
      "className": "vinyl-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <article
              className="album-info"
            >
              <h1
                className="album-name"
              >
                 
                 
              </h1>
              <p
                className="artist"
              >
                 by: 
                 
              </p>
              <p
                className="album-genre"
              >
                 genre: 
                 
              </p>
            </article>,
            <article
              className="track-list"
            >
              <ul>
                 
                 
              </ul>
            </article>,
          ],
          "className": "track-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="album-name"
                >
                   
                   
                </h1>,
                <p
                  className="artist"
                >
                   by: 
                   
                </p>,
                <p
                  className="album-genre"
                >
                   genre: 
                   
                </p>,
              ],
              "className": "album-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    undefined,
                    " ",
                  ],
                  "className": "album-name",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  undefined,
                  " ",
                ],
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " by: ",
                    undefined,
                    " ",
                  ],
                  "className": "artist",
                },
                "ref": null,
                "rendered": Array [
                  " by: ",
                  undefined,
                  " ",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " genre: ",
                    undefined,
                    " ",
                  ],
                  "className": "album-genre",
                },
                "ref": null,
                "rendered": Array [
                  " genre: ",
                  undefined,
                  " ",
                ],
                "type": "p",
              },
            ],
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul>
                 
                 
              </ul>,
              "className": "track-list",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  Array [],
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                " ",
              ],
              "type": "ul",
            },
            "type": "article",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " add a track:",
            <input
              name="name"
              onChange={[Function]}
              placeholder="name"
              value=""
            />,
            <input
              name="duration"
              onChange={[Function]}
              placeholder="duration"
              value=""
            />,
            <button
              type="submit"
            >
               submit 
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          " add a track:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "onChange": [Function],
              "placeholder": "name",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "duration",
              "onChange": [Function],
              "placeholder": "duration",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " submit ",
              "type": "submit",
            },
            "ref": null,
            "rendered": " submit ",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "View All Ablums",
          "className": "view-albums-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "View All Ablums",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="track-container"
          >
            <article
              className="album-info"
            >
              <h1
                className="album-name"
              >
                 
                 
              </h1>
              <p
                className="artist"
              >
                 by: 
                 
              </p>
              <p
                className="album-genre"
              >
                 genre: 
                 
              </p>
            </article>
            <article
              className="track-list"
            >
              <ul>
                 
                 
              </ul>
            </article>
          </section>,
          <form
            onSubmit={[Function]}
          >
             add a track:
            <input
              name="name"
              onChange={[Function]}
              placeholder="name"
              value=""
            />
            <input
              name="duration"
              onChange={[Function]}
              placeholder="duration"
              value=""
            />
            <button
              type="submit"
            >
               submit 
            </button>
          </form>,
          <button
            className="view-albums-button"
            onClick={[Function]}
          >
            View All Ablums
          </button>,
        ],
        "className": "vinyl-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <article
                className="album-info"
              >
                <h1
                  className="album-name"
                >
                   
                   
                </h1>
                <p
                  className="artist"
                >
                   by: 
                   
                </p>
                <p
                  className="album-genre"
                >
                   genre: 
                   
                </p>
              </article>,
              <article
                className="track-list"
              >
                <ul>
                   
                   
                </ul>
              </article>,
            ],
            "className": "track-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="album-name"
                  >
                     
                     
                  </h1>,
                  <p
                    className="artist"
                  >
                     by: 
                     
                  </p>,
                  <p
                    className="album-genre"
                  >
                     genre: 
                     
                  </p>,
                ],
                "className": "album-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      undefined,
                      " ",
                    ],
                    "className": "album-name",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    undefined,
                    " ",
                  ],
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " by: ",
                      undefined,
                      " ",
                    ],
                    "className": "artist",
                  },
                  "ref": null,
                  "rendered": Array [
                    " by: ",
                    undefined,
                    " ",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " genre: ",
                      undefined,
                      " ",
                    ],
                    "className": "album-genre",
                  },
                  "ref": null,
                  "rendered": Array [
                    " genre: ",
                    undefined,
                    " ",
                  ],
                  "type": "p",
                },
              ],
              "type": "article",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ul>
                   
                   
                </ul>,
                "className": "track-list",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    Array [],
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  " ",
                ],
                "type": "ul",
              },
              "type": "article",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " add a track:",
              <input
                name="name"
                onChange={[Function]}
                placeholder="name"
                value=""
              />,
              <input
                name="duration"
                onChange={[Function]}
                placeholder="duration"
                value=""
              />,
              <button
                type="submit"
              >
                 submit 
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            " add a track:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "name",
                "onChange": [Function],
                "placeholder": "name",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "duration",
                "onChange": [Function],
                "placeholder": "duration",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " submit ",
                "type": "submit",
              },
              "ref": null,
              "rendered": " submit ",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "View All Ablums",
            "className": "view-albums-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "View All Ablums",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
